trunk/SUAVE/Methods/Aerodynamics/Common/Fidelity_Zero/Lift/VLM.py

```diff
@@ -305,42 +443,152 @@ def VLM(conditions,settings,geometry):
     X      = VD.XCH[LE_ind]  # These are all LE values
-    #Y      = VD.YCH[LE_ind]  # These are all LE values
+    Y      = VD.YCH[LE_ind]  # These are all LE values
     Z      = VD.ZCH[LE_ind]  # These are all LE values
-    XBAR   = np.ones(sum(LE_ind)) * x_m
-    ZBAR   = np.ones(sum(LE_ind)) * z_m
-    #BMX    = BFZ * Y - BFY * (Z - ZBAR)
+    BMX    = BFZ * Y - BFY * (Z - ZBAR)
+    BMX    = BMX + SICPLE
     BMY    = BMLE * COD + BFX * (Z - ZBAR) - BFZ * (X - XBAR)
-    CDC    = BFZ * SINALF +  BFX * COSALF
+    BMZ    = BMLE * SID - BFX * Y + BFY * (X - XBAR)
+    CDC    = BFZ * SINALF +  (BFX *COPSI + BFY *SINPSI) * COSALF
     CDC    = CDC * CHORD_strip
 
-    ES    = 2*s[0,LE_ind]
-    STRIP = ES *CHORD_strip
-    LIFT  = (BFZ *COSALF - BFX *SINALF)*STRIP
-    DRAG  = CDC*ES 
-
-    MOMENT = STRIP *BMY
+    ES     = 2*s[0,LE_ind]
+    STRIP  = ES *CHORD_strip
+    LIFT   = (BFZ *COSALF - (BFX *COPSI + BFY *SINPSI) *SINALF)*STRIP
+    DRAG   = CDC*ES 
+    MOMENT = STRIP * (BMY *COPSI - BMX *SINPSI)  
+    FY     = (BFY *COPSI - BFX *SINPSI) *STRIP
+    RM     = STRIP *(BMX *COSALF *COPSI + BMY *COSALF *SINPSI + BMZ *SINALF)
+    YM     = STRIP *(BMZ *COSALF - (BMX *COPSI + BMY *SINPSI) *SINALF)
 
-    # Now calculate the coefficients for each wing and in total
+    # Now calculate the coefficients for each wing
     cl_y     = LIFT/CHORD_strip/ES
     cdi_y    = DRAG/CHORD_strip/ES
     CL_wing  = np.add.reduceat(LIFT,span_breaks,axis=1)/SURF
     CDi_wing = np.add.reduceat(DRAG,span_breaks,axis=1)/SURF
-    CL       = np.atleast_2d(np.sum(LIFT,axis=1)/SREF).T
-    CDi      = np.atleast_2d(np.sum(DRAG,axis=1)/SREF).T
-    CM       = np.atleast_2d(np.sum(MOMENT,axis=1)/SREF).T/c_bar
-
     alpha_i = np.hsplit(np.arctan(cdi_y/cl_y),span_breaks[1:])
     
-    return CL, CDi, CM, CL_wing, CDi_wing, cl_y, cdi_y, alpha_i, CP, gamma
\ No newline at end of file
+    # Now calculate total coefficients
+    CL       = np.atleast_2d(np.sum(LIFT,axis=1)/SREF).T          # CLTOT in VORLAX
+    CDi      = np.atleast_2d(np.sum(DRAG,axis=1)/SREF).T          # CDTOT in VORLAX
+    CM       = np.atleast_2d(np.sum(MOMENT,axis=1)/SREF).T/c_bar  # CMTOT in VORLAX
+
+    CYTOT    = np.atleast_2d(np.sum(FY,axis=1)/SREF).T   # total y force coeff
+    CRTOT    = np.atleast_2d(np.sum(RM,axis=1)/SREF).T   # total rolling moment coeff
+    CRMTOT   = CRTOT/w_span*(-1)                         # an output in VORLAG.LOG
+    CNTOT    = np.atleast_2d(np.sum(YM,axis=1)/SREF).T   # total yawing  moment coeff
+    CYMTOT   = CNTOT/w_span*(-1)                         # an output in VORLAG.LOG
+
+    # ---------------------------------------------------------------------------------------
+    # STEP 13: Pack outputs
+    # ------------------ --------------------------------------------------------------------     
+    precision      = settings.floating_point_precision
+    
+    #VORLAX _TOT outputs
+    results = Data()
```

Comment:
There seem to be more outputs here than indicated in the docstring.

Reply:
I'll fix that

Reply:
fixed